### PR TITLE
Small decks for the big library view

### PR DESCRIPTION
**Pull request by bestdani**
The decks_small_left_small.xml and decks_right_small.xml are currently not used just present for switching between a one row a two row layout when four decks are used (flip the comments in the skin.xml section to test it).

### CODE REVIEW

**Comment by sblaisot** [comment]
you probably didn't want to add a file with "Kopie" in it, right ?
res/skins/LateNight/btn_volume_display_sampler_over - Kopie.png

File: res/skins/LateNight/skin.xml [hunk 1]
```diff
@@ -106,17 +107,26 @@
...
+									<Template src="skin:decks_left_small.xml"/>
+									<Template src="skin:decks_right_small.xml"/>
+									-->
+									<Template src="skin:decks_small_horizontal.xml"/>
+								</Children>
+                            </WidgetGroup>	<SingletonContainer>
                                 <ObjectName>LibrarySingleton</ObjectName>
-                            </SingletonContainer>
-                            <Template src="skin:toolbar_big_library.xml"/>
+                            </SingletonContainer>
```
Comment (sblaisot) [inline res/skins/LateNight/skin.xml]: extra end of line spaces here

File: res/skins/LateNight/skin.xml [hunk 2]
```diff
@@ -106,17 +107,26 @@
...
+								</Children>
+                            </WidgetGroup>	<SingletonContainer>
                                 <ObjectName>LibrarySingleton</ObjectName>
-                            </SingletonContainer>
-                            <Template src="skin:toolbar_big_library.xml"/>
+                            </SingletonContainer>		
+							<Template src="skin:toolbar_big_library.xml"/>
                         </Children>
-                    </WidgetGroup>
+                    </WidgetGroup>
```
Comment (sblaisot) [inline res/skins/LateNight/skin.xml]: extra and-of-line spaces

File: res/skins/LateNight/skin.xml [hunk 3]
```diff
@@ -106,17 +107,26 @@
...
+									<Template src="skin:decks_right_small.xml"/>
+									-->
+									<Template src="skin:decks_small_horizontal.xml"/>
+								</Children>
+                            </WidgetGroup>	<SingletonContainer>
                                 <ObjectName>LibrarySingleton</ObjectName>
-                            </SingletonContainer>
-                            <Template src="skin:toolbar_big_library.xml"/>
+                            </SingletonContainer>		
+							<Template src="skin:toolbar_big_library.xml"/>
```
Comment (sblaisot) [inline res/skins/LateNight/skin.xml]: indentation mismatch ? (maybe spaces versus tabs)

File: res/skins/LateNight/style.qss [hunk 4]
```diff
@@ -20,6 +20,31 @@
   qproperty-layoutAlignment: 'AlignHCenter | AlignTop';
 }
 
+#DeckSmall3{
+  qproperty-layoutAlignment: 'AlignHCenter | AlignTop';
+}
+
+#DeckSmall4 {
+  qproperty-layoutAlignment: 'AlignHCenter | AlignTop';
+  border: 1px solid #585858;
```
Comment (sblaisot) [inline res/skins/LateNight/style.qss]: why are decks 1/3 without border and deck 2/4 with a right border ?

File: res/skins/LateNight/decks_small_horizontal.xml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                        </Children>
+                    </WidgetGroup>
+                    <WidgetGroup>
+                        <ObjectName>Deck</ObjectName>
+                        <Layout>horizontal</Layout>
+                        <SizePolicy>me,me</SizePolicy>
+                        <Children>
+                            <Template src="skin:deck_small.xml">
+								<SetVariable name="channum">2</SetVariable>
+								<SetVariable name="signal_color">#E7C413</SetVariable>
```
Comment (sblaisot) [inline res/skins/LateNight/decks_small_horizontal.xml]: indentation problem (tab vs space)

File: res/skins/LateNight/decks_small_horizontal.xml [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+                            <BindProperty>visible</BindProperty>
+                        </Connection>
+                    </WidgetGroup>
+                    <WidgetGroup>
+                        <ObjectName>Deck</ObjectName>
+                        <Layout>horizontal</Layout>
+                        <SizePolicy>me,me</SizePolicy>
+                        <Children>
+                            <Template src="skin:deck_small.xml">
+								<SetVariable name="channum">1</SetVariable>
```
Comment (sblaisot) [inline res/skins/LateNight/decks_small_horizontal.xml]: indentation here too (tab vs space)

File: res/skins/LateNight/decks_right_small.xml [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        <Layout>vertical</Layout>
+        <SizePolicy>me,min</SizePolicy>
+        <Children>
+            <WidgetGroup>
+                <ObjectName>DecksRightContainer</ObjectName>
+                <Layout>vertical</Layout>
+                <SizePolicy>me,min</SizePolicy>
+                <Children>
+                    <Template src="skin:deck_small.xml">
+        	            <SetVariable name="channum">2</SetVariable>
```
Comment (sblaisot) [inline res/skins/LateNight/decks_right_small.xml]: extra indentation on this line

File: res/skins/LateNight/decks_left_small.xml [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        <Layout>vertical</Layout>
+        <SizePolicy>me,min</SizePolicy>
+        <Children>
+            <WidgetGroup>
+                <ObjectName>DecksLeftContainer</ObjectName>
+                <Layout>vertical</Layout>
+                <SizePolicy>me,min</SizePolicy>
+                <Children>
+                    <Template src="skin:deck_small.xml">
+        	            <SetVariable name="channum">1</SetVariable>
```
Comment (sblaisot) [inline res/skins/LateNight/decks_left_small.xml]: extra indentation here

File: res/skins/LateNight/deck_small_row_1.xml [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+								<!--<Elide>right</Elide>-->
+								<SizePolicy>min,min</SizePolicy>
+								<Channel>
+									<Variable name="channum" />
+								</Channel>
+							</TrackProperty>
+							<Label>
+								<ObjectName>TextColor<Variable name="channum" /></ObjectName>
+								<Style>QLabel { font: bold 13px/15px Lucida Grande, Lucida Sans
+									Unicode, Arial, Verdana, sans-serif; text-align: left;}</Style>
```
Comment (sblaisot) [inline res/skins/LateNight/deck_small_row_1.xml]: Please add "Segoe UI" for windows users. font: lines should be consistent across whole skin.

File: res/skins/LateNight/deck_small.xml [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+                                  background-color: transparent; padding: 0px; margin: 0px;}
+                              </Style>
+                              <Group>[Channel<Variable name="channum"/>]</Group>
+                              <SizePolicy>min,f</SizePolicy>
+                              <Size>-1,12</Size>
+                              <Connection>
+                                <ConfigKey>[Channel<Variable name="channum"/>],visual_key</ConfigKey>
+                              </Connection>
+                           </Key>
+							<WidgetGroup>
```
Comment (sblaisot) [inline res/skins/LateNight/deck_small.xml]: indentation: tabs vs spaces

**Comment by sblaisot** [comment]
can you please:
* keep indentation with 4 spaces on all files: https://www.mixxx.org/wiki/doku.php/coding_guidelines#tabs_vs_spaces
* make `font:` elements consistent with rest of the skin

Thanks.

sb

**Comment by rryan** [comment]
Does this PR supersede #677 and #870 ?

**Comment by Be-ing** [comment]
This pull request has not been updated in years so I am closing it to reduce the clutter of open PRs. If you solve the merge conflicts and would like to have this merged in the future, please reopen it. However, I am not convinced anymore that this is a great idea. To me it feels like a hack around around LateNight's waste of vertical space which has since been solved.

**Comment by daschuer** [comment]
There is still a demand for small decks in a Big library.

**Comment by Be-ing** [comment]
Is there really? Why? Who is asking for it?

**Comment by daschuer** [comment]
It was important enough that three contributors spend their spare time.

**Comment by Be-ing** [comment]
But that was before @ronso0 redesigned LateNight to not waste so much vertical space.

**Comment by ronso0** [comment]
Minimal decks would be a nice feature when working with a maximized libbrary, and -IMO- are independent from how much space decks, mixer etc. need in regular use.
Here's the bug @Be-ing opened https://bugs.launchpad.net/mixxx/+bug/1467004

This is on my ToDo for Tango and other skins as well.
Let's see how it works out here, or keep it open as reminder/reference in case it's stalled. Probably we can adapt things for minimal LateNight samplers as well.

**Comment by ronso0** [comment]
new attempt in #1933

**Comment by ronso0** [comment]
This can be closed as the #1933 implementation is merged.

**Pull request closed without merging**